src/ApiService/ApiService/onefuzzlib/JobResultOperations.cs

```diff
@@ -0,0 +1,126 @@
+    }
+
+    public async Async.Task<JobResult?> GetJobResult(Guid jobId) {
+        return await SearchByPartitionKeys(new[] { jobId.ToString() }).SingleOrDefaultAsync();
+    }
+
+    private JobResult UpdateResult(JobResult result, JobResultType type, Dictionary<string, double> resultValue) {
+
+        var newResult = result;
+        double newValue;
+        switch (type) {
+            case JobResultType.NewCrashingInput:
+                newValue = result.NewCrashingInput + resultValue["count"];
+                newResult = result with { NewCrashingInput = newValue };
+                break;
+            case JobResultType.NewReport:
+                newValue = result.NewReport + resultValue["count"];
+                newResult = result with { NewReport = newValue };
+                break;
+            case JobResultType.NewUniqueReport:
+                newValue = result.NewUniqueReport + resultValue["count"];
+                newResult = result with { NewUniqueReport = newValue };
+                break;
+            case JobResultType.NewRegressionReport:
+                newValue = result.NewRegressionReport + resultValue["count"];
+                newResult = result with { NewRegressionReport = newValue };
+                break;
+            case JobResultType.CoverageData:
+                double newCovered = resultValue["covered"];
+                double newTotalCovered = resultValue["features"];
+                double newCoverageRate = resultValue["rate"];
+                newResult = result with { InstructionsCovered = newCovered, TotalInstructions = newTotalCovered, CoverageRate = newCoverageRate };
+                break;
+            case JobResultType.RuntimeStats:
+                _logTracer.LogInformation("Attempting update to iterations.");
+                double newTotalIterations = resultValue["total_count"];
+                _logTracer.LogInformation($"Attempting update to iterations {newTotalIterations}.");
+                // int newExecsSec = resultValue["execs_sec"];
+                newResult = result with { IterationCount = newTotalIterations };
+                break;
+            default:
+                _logTracer.LogInformation($"Invalid Field {type}.");
+                break;
+        }
+        _logTracer.LogInformation($"Attempting to log new result: {newResult}");
+        return newResult;
+    }
+
+    private async Async.Task<bool> TryUpdate(Job job, JobResultType resultType, Dictionary<string, double> resultValue) {
+        var jobId = job.JobId;
+        _logTracer.LogInformation($"Inside try, but before.");
+
+        var jobResult = await GetJobResult(jobId);
+
+        _logTracer.LogInformation($"Inside try.");
+        if (jobResult == null) {
+            _logTracer.LogInformation("Creating new JobResult for Job {JobId}", jobId);
+
+            var entry = new JobResult(JobId: jobId, Project: job.Config.Project, Name: job.Config.Name, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```

Comment:
To simplify the use of JobResult, let's have a constructor that initializes all the values to 0